packages/client/e2e/test-d.spec.ts

```diff
@@ -0,0 +1,403 @@
+    const [{ context: userAContext }, { context: userBContext }] =
+        await Promise.all([
+            createContext({ browser, index: 0 }),
+            createContext({ browser, index: 1 }),
+        ]);
+    const [{ page: userADevice1Page }, { page: userBPage }] = await Promise.all(
+        [
+            setupPageFromContext(userAContext),
+
+            setupPageFromContext(userBContext),
+        ],
+    );
+
+    const roomName = 'MusicRoom is the best';
+
+    await createPublicRoomWithInvitation({ page: userADevice1Page, roomName });
+
+    const { page: userADevice2Page } = await setupPageFromContext(userAContext);
+    await waitForJoiningRoom({ page: userADevice2Page, roomName });
+
+    const { page: userADevice3Page } = await setupPageFromContext(userAContext);
+    await waitForJoiningRoom({ page: userADevice3Page, roomName });
+
+    await joinRoom({ page: userBPage, roomName });
+
+    await Promise.all([
+        waitForYouTubeVideoToLoad(userADevice1Page),
+        waitForYouTubeVideoToLoad(userADevice2Page),
+        waitForYouTubeVideoToLoad(userADevice3Page),
+        waitForYouTubeVideoToLoad(userBPage),
+    ]);
+
+    await Promise.all([
+        playTrack(userADevice2Page),
+
+        waitForPlayerState({
+            page: userADevice1Page,
+            testID: 'music-player-playing-device-emitting',
+        }),
+        waitForPlayerState({
+            page: userADevice2Page,
+            testID: 'music-player-playing-device-muted',
+        }),
+        waitForPlayerState({
+            page: userADevice3Page,
+            testID: 'music-player-playing-device-muted',
+        }),
+        waitForPlayerState({
+            page: userBPage,
+            testID: 'music-player-playing-device-emitting',
+        }),
+    ]);
+
+    await Promise.all([
+        changeEmittingDevice({
+            page: userADevice1Page,
+            emittingDeviceIndex: 1,
+        }),
+
+        waitForPlayerState({
+            page: userADevice1Page,
+            testID: 'music-player-playing-device-muted',
+        }),
+        waitForPlayerState({
+            page: userADevice2Page,
+            testID: 'music-player-playing-device-emitting',
+        }),
+        waitForPlayerState({
+            page: userADevice3Page,
+            testID: 'music-player-playing-device-muted',
+        }),
+        waitForPlayerState({
+            page: userBPage,
+            testID: 'music-player-playing-device-emitting',
+        }),
+    ]);
+
+    await Promise.all([
+        userADevice2Page.close(),
+
+        waitForPlayerState({
+            page: userADevice1Page,
+            testID: 'music-player-playing-device-emitting',
+        }),
+        waitForPlayerState({
+            page: userADevice3Page,
+            testID: 'music-player-playing-device-muted',
+        }),
+        waitForPlayerState({
+            page: userBPage,
+            testID: 'music-player-playing-device-emitting',
+        }),
+    ]);
```

Comment:
Unless I'm mistaken, the userB getting the forcedDisconnection modal is not tested ?